src/api/Api.ts

```diff
@@ -14,6 +14,8 @@ import {
     RoutingProfile,
 } from '@/api/graphhopper'
 import { LineString } from 'geojson'
+import { getTranslation } from '@/translation/Translation'
+const tr = getTranslation()
```

Comment:
This is not ideal, because just loading the file executes `getTranslation()` and this fails if there was no `setTranslation()` before. Why do we even need `getTranslation()`? Can we not keep the `Translation` object local to `Translation.ts` and export only the `tr` function instead?

Reply:
Probably a good idea, yes. I played around and wasn't sure how to integrate this properly to make it convenient but also aligned to other code. My biggest problem with it is that this is a singleton but probably this is fine for a web app :)

Reply:
Can you take a look at my last commit to see if you like it?

Reply:
And what is your issue with the singleton? I think `tr.json` should only be loaded once (this data is global so to say). Other than that all we need is some handy access function to retrieve the data. We also need to keep some state to keep track which is the current language. This is handled by the `Translation` object currently and it makes sense there is only one such object?

Reply:
> And what is your concern about the singleton?

I just dislike singletons in general (probably from my previous life as Android dev where it made everything a mess :D )

> Can you take a look at my last commit to see if you like it?

Looks even more convenient, yes. But the mechanism is not really changed. I.e. if setTranslation is not called before the usage you'll still get an error (?)

Reply:
> The mechanism is not really changed though. I.e. if setTranslation is not called before the usage you'll still get an error 

Yes. `setTranslation` still needs to be called before using it (makes sense). The problem was that I put `const tr = getTranslation()` at the beginning of the file. I did this because otherwise I would have to do `getTranslation().get(..)` wherever I wanted to translate something. But on the other hand when put `const tr = getTranslation` at the beginning of the file this is executed as soon as the file is loaded and the order of the import statements becomes relevant which I thought was ugly.

Reply:
Ah, ok :+1: